loki-build-image/Dockerfile

```
@@ -26,10 +26,13 @@ RUN apt-get update && \
 COPY --from=docker /usr/bin/docker /usr/bin/docker
 COPY --from=helm /usr/bin/helm /usr/bin/helm
 COPY --from=golangci /bin/golangci-lint /usr/local/bin
-RUN go get \
-    github.com/golang/protobuf/protoc-gen-go \
-    github.com/gogo/protobuf/protoc-gen-gogoslick \
-    github.com/gogo/protobuf/gogoproto \
+
+# Enable go 1.11 modules to be able to install pinned version of dependencies
+# even if we're installing them in the GOPATH
+RUN GO111MODULE=on go get \
+    github.com/golang/protobuf/protoc-gen-go@v1.3.0 \
+    github.com/gogo/protobuf/protoc-gen-gogoslick@v1.3.0 \
+    github.com/gogo/protobuf/gogoproto@v1.3.0 \
     github.com/go-delve/delve/cmd/dlv \
     golang.org/x/tools/cmd/goyacc \
```

Wondering if we should pin the yacc compiler to avoid similar issues (although it doesn't seem like this gets updated very much)

```suggestion
    golang.org/x/tools/cmd/goyacc@58d531046acdc757f177387bc1725bfa79895d69 \
```
There didn't seem to be proper releases we could use here so I pulled the git hash from the most recent tag:  https://github.com/golang/tools/releases/tag/gopls%2Fv0.1.7

WDYT?

---

I didn't pinned because of the lack of a proper release and unfrequent updates, but looks wis to force a commit hash. Done!